packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -635,6 +636,33 @@ export class RestAPIClient {
 
 	/* Posts */
 
+	/**
+	 * Given a siteID, checks whether any posts exists of a given state.
+	 *
+	 * @param {number} siteID Site ID.
+	 * @param param1 Keyed object parameter.
+	 * @param {SitePostState} param1.state State of the published post.
+	 */
+	async siteHasPost(
+		siteID: number,
+		{ state = 'publish' }: { state: SitePostState }
+	): Promise< boolean > {
+		const params: RequestParams = {
+			method: 'get',
+			headers: {
+				Authorization: await this.getAuthorizationHeader( 'bearer' ),
+				'Content-Type': this.getContentTypeHeader( 'json' ),
+			},
+		};
+
+		const response = await this.sendRequest(
+			this.getRequestURL( '1.1', `/sites/${ siteID }/post-counts/post` ),
+			params
+		);
+
+		return response.counts.all[ state ] !== 0 ? true : false;
```

Comment:
**suggestion (non-blocking):** We can simplify this to `return response.counts.all[state] !== 0` since it already returns a boolean value.

Reply:
Haha, what an embarrassingly unnecessary ternary.

Reply:
We've all been there! 😄